### DISCUSSION TITLE
MS SQL Server schema question

### DISCUSSION THREAD

**Discussion by person1534** [Budibase/budibase#10008] [Help]
Hey, guys,

I've got 2 tables with the same name in one  mssql database. The only difference is a schema name. Something like "dbo.TABLE1" and "user.TABLE1". The tables structure is kind of similar but totally not same. The Data Adpter has a [schema] setting which led me to an idea you are able to distinguish those. Well, that doesn't happen in practice. I can successfully register say "dbo.TABLE1". But once "user.TABLE1" comes into a picture everything gets messed up. A registered version of "user.TABLE1" has a doubled primary key and column definitions of "dbo.TABLE1". So, it fails to fetch any rows etc. etc. etc. Just frustrating red color popups. A secret recipe here is to rename a table "user.TABLE1" to something else. But eventually a [schema] setting for MS SQL Data Provider looks to have no meaning.

Best regards, Dmitry

**Comment by melohagan** [comment]
Could you provide screenshots please? I'm not quite following the issue here!

**Reply by gmentzik** [reply]
I use table prefixes in db to avoid messing app table names.
